code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -90,27 +90,46 @@
 			P.build_pipeline(src)
 
 /obj/machinery/atmospherics/components/proc/nullifyPipenet(datum/pipeline/reference)
+	// Crash when we call a null pipenet here. Pretty straightforward.
 	if(!reference)
 		CRASH("nullifyPipenet(null) called by [type] on [COORD(src)]")
-	var/i = parents.Find(reference)
-	reference.other_airs -= airs[i]
+
+	// Disconnect each air mixtures from the other_airs list in the pipeline. 
+	// Also nullifies the specific entry in the parents list. This is mainly used when a machinery is rotated.
+	for (var/i in 1 to parents.len)
+		if (parents[i] == reference)
+			reference.other_airs -= airs[i]
+			parents[i] = null
+
+	// Deletes any reference of the atmos machinery as listed in the other_atmosmch list from the pipeine.
 	reference.other_atmosmch -= src
+
+	// Cleans pipenet up
+	if(!length(reference.other_atmosmch) && !length(reference.members))
+		if(QDESTROYING(reference))
+			CRASH("nullifyPipenet() called on qdeleting [reference]")
+		qdel(reference)
+
 	/**
 	 *  We explicitly qdel pipeline when this particular pipeline
 	 *  is projected to have no member and cause GC problems.
 	 *  We have to do this because components don't qdel pipelines
 	 *  while pipes must and will happily wreck and rebuild everything
 	 *	again every time they are qdeleted.
 	 */
```

Comment:
Code here looks great, I think you're falling for a trap when it comes to the comments tho.

If a comment can be replaced by just reading the code, it shouldn't exist. Code should in theory be self documenting, and if the comment isn't saying anything unique it's just adding visual noise, not to mention the chance of it going out of date.

Optimally comments should describe why we're doing something, and only detail what exactly we're doing if it's especially confusing.

Oh and you should move the comment about qdeling pipelines to above the bit that qdels pipelines, to make things more clear.

Reply:
I moved the comment up and removed the unecessary comments. However i wanted to keep the short description on nullifyPipenet, hope this is okay. 
Also i mistyped pipeline, be right back.

Reply:
Oh of course, comments aren't bad, and these are just fine. Just be weary of overdoing it. Don't let me discourage you from documenting your code, just keep why in mind rather then how